### PR TITLE
Fixed Gaussian curve Peak Transfer overfitting

### PR DESCRIPTION
**Pull request by Mailaender**
https://github.com/OpenChrom/openchrom/issues/556 already improved the situation because it adjusts the narrowness of the curves, but we can do better by trimming sigma even further until it fits under the total signal. This optimization function takes some time, so I added a progress monitor. Step size is also optimized for speed and could be reduced for better fits, but it didn't matter too much in my test data.

### CODE REVIEW

File: openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/peaks/PeakTransfer.java [hunk 1]
```diff
@@ -272,7 +274,7 @@ private void transfer(IPeak peakSource, int startRetentionTime, int stopRetentio
 		//
 		IPeak peakSink = peakSupport.extractPeakByRetentionTime(chromatogramSink, startRetentionTime, stopRetentionTime, includeBackground, optimizeRange, traces);
 		if(peakSink != null) {
-			adjustPeakIntensity(peakSource, peakSink, percentageIntensity, peakTransferSettings);
+			adjustPeakIntensity(peakSource, percentageIntensity, peakTransferSettings);
```
Comment (eselmeister) [inline openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/peaks/PeakTransfer.java]: In this case, the intensity of the source peak is adjusted. I doubt, that this is done on purpose.

File: openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/peaks/PeakTransfer.java [hunk 2]
```diff
@@ -333,7 +323,40 @@ private void transfer(IPeak peakSource, IChromatogram<? extends IPeak> chromatog
...
+				return null;
+			}
+		}
+		return peakSink;
+	}
+
+	private boolean isWithinBounds(IPeak peakSink, IChromatogramCSD chromatogramCSD) {
+
+		IPeakModel peakModel = peakSink.getPeakModel();
+		for(int rt : peakModel.getRetentionTimes()) {
```
Comment (eselmeister) [inline openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/peaks/PeakTransfer.java]: using rt is fine, but I prefer to use the full word, here: `for(int retentionTime : peakModel.getRetentionTimes()) {`

**Comment by eselmeister** [comment]
Don't forget to put yourself in the header.

File: openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/peaks/PeakTransfer.java (hunk 1 above)
Comment (Mailaender) [reply openchrom/plugins/net.openchrom.xxd.process.supplier.templates/src/net/openchrom/xxd/process/supplier/templates/peaks/PeakTransfer.java]: Whoopsy. Well spotted.

**Pull request merged**